### ISSUE TITLE
New addon which adds focus rings to scratch-www buttons

### ISSUE THREAD

**Issue by towerofnix** [ScratchAddons/ScratchAddons#4032]
### Is your request related to a problem? Please describe.

Right now using the tab key to switch between buttons without the mouse is kind of awkward, because scratch-www pages don't put focus rings (outlines showing what is focused) on buttons.

### Describe the solution you'd like

I'd like an addon which shows a nice, clearly visible focus ring around the focused button, using the CSS `:focus` selector to target. Ideally, the same (or similar) style should be applied to 2.0 -> 3.0 buttons (which have a dotted outline focus ring but it's still kinda hard to see), and it should be clearly visible (maybe different color) on both light and dark WWW. (This might be a little trickier to write dynamically now that the dark WWW addon has full customization support.)

### Describe alternatives you've considered

Other alternatives include e.g. making the focused button darker, instead of adding a focus ring. This is *alright,* but focus rings are usually easier to visually recognize.

GitHub buttons, just for example, use the actual color of the button (like green in "Submit new issue") to color the focus ring (which is a transparent version of that color). You'd probably need to use JS instead of only a simple userstyle to get that working, but it might ease the trouble of making the focus ring look good regardless of user customization, especially if the user customizes the highlight color (button color) in the customizable WWW colors addon.

### Additional context

The addon itself isn't fully accessible with just tabbing etc (see #2532), but neither is Scratch (especially the editor). Still, parts of the site (like commenting) can be accessed without the mouse, so I think improving support for that would be nice (and useful!).

**Comment by mxmou** [comment]
This could be very useful. I think the CSS `:focus-visible` selector should be used to only show the outline when the button is focused using the keyboard.

> GitHub buttons, just for example, use the actual color of the button (like green in "Submit new issue") to color the focus ring (which is a transparent version of that color). You'd probably need to use JS instead of only a simple userstyle to get that working, but it might ease the trouble of making the focus ring look good regardless of user customization, especially if the user customizes the highlight color (button color) in the customizable WWW colors addon.

That's probably the best option because it fits with Scratch's design (for example the selected sprite outline in the editor).
![image](https://user-images.githubusercontent.com/51849865/148391229-9000aeba-471e-4d5f-b6f8-d97372050651.png)
I don't think any JavaScript is necessary to implement it - few buttons don't use the highlight color, so specific CSS for them could be added.

**Comment by towerofnix** [comment]
>  I don't think any JavaScript is necessary to implement it - few buttons don't use the highlight color, so specific CSS for them could be added.

One possible catch is that CSS colors aren't really first class the same way numbers are—you can't alter them using operators like `calc()`. So, you can't take a `#rrggbb` formatted value and convert that to an `rgba` value (with arbitrarily provided opacity). The userstyle settings would have to specifically generate a `--ring-color` value which is the same as highlight color but partially transparent.

(You'd also have to make sure this is generated by default by *this* addon too, the one adding focus rings, so that focus rings still work without customizable WWW—and arrange this addon to be lower priority than customizable WWW, so its value overrides the default from this addon.)
